openssl/src/ec.rs

```diff
@@ -1009,6 +1009,26 @@ impl EcKey<Private> {
         EcKey<Private>,
         ffi::d2i_ECPrivateKey
     }
+
+    /// Decodes a DER-encoded elliptic curve private key structure for the specified curve.
+    #[corresponds(EC_KEY_parse_private_key)]
+    #[cfg(boringssl)]
+    pub fn private_key_from_der_for_group(
+        der: &[u8],
+        group: &EcGroupRef,
+    ) -> Result<EcKey<Private>, ErrorStack> {
+        unsafe {
+            let mut cbs = ffi::CBS {
+                data: der.as_ptr(),
+                len: der.len(),
+            };
+            cvt_p(ffi::EC_KEY_parse_private_key(
+                &mut cbs as *mut ffi::CBS,
+                group.as_ptr(),
+            ))
+            .map(|p| EcKey::from_ptr(p))
```

Comment:
Without commenting on whether this makes sense I'm this project or not, FYI this patch is wrong. It doesn't account for trailing data by checking the CBS is empty after parsing.

Reply:
(There also is a way to do this in OpenSSL but it's suuuuuuuper confusing and hard to use. `d2i_ECPrivateKey` peeks at the state of the input `EC_KEY` in the object reuse mode. You also have to be very careful when using the object reuse mode because the function might destroy the object and put a different one in its place.)

Reply:
Yup. Both CBS and d2i functions parse a prefix of the input and tell you how far they got, so that you can parse a series of things. I'm... not sure this was the least errorprone API design in hindsight but here we are. This means that if you are implementing an API that just parses it, rather than parses and tells you what was left, you need to check the input was fully consumed.

Reply:
Gotcha; does the new version I uploaded look good? (Assuming we figure out a reasonable error to report)